code/scripting/api/libs/ui.cpp

```diff
@@ -542,6 +556,225 @@ ADE_FUNC(getCmdBriefing,
 	return ade_set_args(L, "o", l_CmdBrief.Set(Cmd_briefs[0]));
 }
 
+//**********SUBLIBRARY: UserInterface/Debriefing
+ADE_LIB_DERIV(l_UserInterface_Debrief,
+	"Debriefing",
+	nullptr,
+	"API for accessing data related to the debriefing UI.<br><b>Warning:</b> This is an internal "
+	"API for the new UI system. This should not be used by other code and may be removed in the future!",
+	l_UserInterface);
+
+ADE_FUNC(initDebriefing,
+	l_UserInterface_Debrief,
+	nullptr,
+	"Builds the debriefing, the stats, sets the next campaign mission, and makes all relevant data accessible",
+	"number",
+	"Returns 1 when completed")
+{
+	//This is used to skip some UI preloading in debrief init
+	API_Access = true;
+	
+	//freespace.cpp does all this stuff before entering debrief so we have to do it here, too -Mjn
+	hud_stop_looped_locking_sounds();
+	hud_stop_looped_engine_sounds();
+	afterburner_stop_sounds();
+	player_stop_looped_sounds();
+	obj_snd_stop_all(); // stop all object-linked persistant sounds
+	if (Subspace_ambient_left_channel.isValid()) {
+		snd_stop(Subspace_ambient_left_channel);
+		Subspace_ambient_left_channel = sound_handle::invalid();
+	}
+
+	if (Subspace_ambient_right_channel.isValid()) {
+		snd_stop(Subspace_ambient_right_channel);
+		Subspace_ambient_right_channel = sound_handle::invalid();
+	}
+	snd_stop(Radar_static_looping);
+	Radar_static_looping = sound_handle::invalid();
+	snd_stop(Target_static_looping);
+	shipfx_stop_engine_wash_sound();
+	Target_static_looping = sound_handle::invalid();
```

Comment:
These lines should all be replaced with a call to `game_stop_looped_sounds()`.

Reply:
This exists in freespace.h which is the reason used all of these here instead. From what I understand, I can't call a method that exists outside of Code.

Reply:
No, it's still possible; `game_stop_looped_sounds()` is called from keycontrol.cpp.

There's no inherent restriction against calling functions between projects, but there are occasional gotchas.  The main gotcha for our purposes is that any function in freespace.h must have a "stub function" that is defined when you're not compiling freespace.cpp.  This is why FRED and qtFRED have stub functions in their respective copies of fredstubs.cpp.

Reply:
Alright